Week3/homework/2-step3.js

```diff
@@ -1,8 +1,41 @@
 'use strict';
+let range = function (startIndex ,stopIndex) {
+  const values = [];
+  let i = startIndex;
+  while(i <= stopIndex){
+    values.push(i);
+    i++;
+  }
+  return values;
+
+}
+function sayThree(){
+  return "sayThree";
+}
+
+function sayFive(){
+  return "sayFive";
+}
 
 function threeFive(startIndex, stopIndex, threeCallback, fiveCallback) {
-  const values = [];
-  // Add your code here
+
+let newValues = range (startIndex,stopIndex);
+let result = [];
+for (let i = 0; i < newValues.length; i++){
+  if (newValues[i] % 5 === 0 && newValues[i] % 3 === 0){
```

Comment:
There is no need for this complicated `if-else` construct. Testing for divisibility by 3 and 5 independently will do the same:

```js
if (newValues[i] % 3 === 0) {
  result.push(threeCallback());
}

if (newValues[i] % 5 === 0) {
  result.push(fiveCallback());
}
```